src/main/java/com/kwlz/rms/repository/model/UserRoomKey.java

```diff
@@ -0,0 +1,29 @@
+package com.kwlz.rms.repository.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import java.io.Serializable;
+import java.math.BigDecimal;
+
+@Data
+@Builder
+@Embeddable
+@NoArgsConstructor
+@AllArgsConstructor
+public class UserRoomKey implements Serializable {
+
+    @Column(name = "user_id", nullable = false)
+    private BigDecimal userId;
+
+    @Column(name = "room_id", nullable = false)
+    private BigDecimal roomId;
+
+    @Column(name = "room_status_id", nullable = false)
+    private BigDecimal roomStatusId;
```

Comment:
To tutaj raczej nie powinno być spowoduje że dany pokój może być tylko 1 raz zarezerwowany, a można by to zrobić tak  że użytkownik zarezerwuje sobie ten pokój w różnych terminach. Zostawiłbym tu tylko userID i RoomID a resztę walidował w trakcie działania aplikacji.